### PR TITLE
Fixes Minor Issues in Routes

### PR DESCRIPTION
**Pull request by sdhani**
## Fixes Plant Add/Update Issues in Routes
- [x] `/api/plants/:id` should return an object now (and not an array)
- [x] `outdoor_plant` should be modified during add plant `POST`
- [x] added `name` field to plant :sparkles:

Implementations Requires Local Changes:
- `dropdb plant-pal` (or `dropdb plantpal`)
- `createdb plant-pal` (`createdb plantpal`)
- `knex migrate:latest`
> Note: Seed data is OUT-OF-DATE. Avoid use.

### CODE REVIEW

File: routes/RESTful_API_Docs.md [hunk 1]
```diff
@@ -1,42 +1,42 @@
...
-| plants | CRUD, Query | yes| Completed |
-| gardens | CRUD | yes | Completed |
-| weather | TBD | TBD |W.I.P| 
-| location | TBD | TBD |W.I.P| 
+| :------------: | :------------------: | :------------: | :---------: |
+|  users [auth]  |  CRUD, Login/Logout  |      yes       |  Completed  |
+|     plants     |     CRUD, Query      |      yes       |  Completed  |
+|    gardens     |         CRUD         |      yes       |  Completed  |
+|    weather     |         TBD          |      TBD       |    W.I.P    |
+|    location    |         TBD          |      TBD       |    W.I.P    |
```
Comment (benevolentPreta) [inline routes/RESTful_API_Docs.md:21]: I'll drop this (location) table when I do my documentation later.

File: routes/RESTful_API_Docs.md [hunk 2]
```diff
@@ -1,42 +1,42 @@
...
 A request with the proper body must be sent out to: `/api/*database_table*/*operation*`
- 
 
 **Examples:** [POST]: `localhost:3001/api/auth/register`
+
 ```javascript
 req.body = {
-  "email":"awesome148@people.com",
-  "display_name":"Awesome148",
-  "zipcode":"10000",
```
Comment (benevolentPreta) [inline routes/RESTful_API_Docs.md:34]: I'll change this too, example zip should be valid. I'll just use Hunters 10065

File: routes/RESTful_API_Docs.md [hunk 3]
```diff
@@ -274,33 +306,37 @@ req.body = {
...
-|  | images | json |
-|  | foliage | json |
-|  | fruit_or_seed | json |
-|  | growth | json |
-|  | seed | json |
-|  | specifications | json |
-|  | family_common_name | varchar(255) |
+
+| Key |       Column       |     Type      |
+| :-: | :----------------: | :-----------: |
```
Comment (benevolentPreta) [inline routes/RESTful_API_Docs.md:320]: "name" (user specific name), should be added to this now?

Comment (sdhani) [reply routes/RESTful_API_Docs.md:320]: good catch!

**Pull request merged**